index.js

```diff
@@ -52,7 +51,9 @@ const preprocessDestinationPath = (source, destination, options) => {
 	}
 
 	if (options.parents) {
-		return path.join(destination, dirname, basename);
+		const dirname = path.dirname(source);
+		const parsedDirectory = path.parse(dirname);
+		return path.join(destination, dirname.replace(parsedDirectory.root, path.sep), basename);
```

Comment:
`destination` and `dirname` can both be absolute paths.  When this occurs you end up with the following:

```sh
# MacOS & Linux
/tmp/example + /tmp/example/cwd => /tmp/example/tmp/example/cwd

# Windows
C:\\tmp\\example + C:\\tmp\\example\\cwd => C:\\tmp\\example\\C:\\tmp\\example\\cwd
```

The resulting Windows path is invalid.  This strips the root ( `C:\\` ) from the directory path before joining by replacing it with the path seperator.  This will be a NOOP on non-Windows.

Reply:
Good catch. Would you be able to add a test for that?

Reply:
This is covered by the 'path structure' test which was failing prior to this fix.